src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/DiagnosticTest.cs

```diff
@@ -17,21 +17,20 @@
 using System.Runtime.CompilerServices;
 using System;
 using System.Data;
+using Microsoft.DotNet.RemoteExecutor;
 
-namespace Microsoft.Data.SqlClient.Tests
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
 {
-    public class DiagnosticTest : RemoteExecutorTestBase
+    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+    public class DiagnosticTest
     {
         private const string BadConnectionString = "data source = bad; initial catalog = bad; uid = bad; password = bad; connection timeout = 1;";
-        private static readonly string s_tcpConnStr = Environment.GetEnvironmentVariable("TEST_TCP_CONN_STR") ?? string.Empty;
-        
-        public static bool IsConnectionStringConfigured() => s_tcpConnStr != string.Empty;
+        private static readonly string s_tcpConnStr = DataTestUtility.TCPConnectionString ?? string.Empty;
 
-        [Fact]
-        [SkipOnTargetFramework(TargetFrameworkMonikers.UapAot | TargetFrameworkMonikers.NetFramework,  "Internals reflection not supported on UapAot | Feature not available on Framework")]
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
         public void ExecuteScalarTest()
         {
-            RemoteInvoke(() =>
+            RemoteExecutor.Invoke(() =>
             {
                 CollectStatisticsDiagnostics(connectionString =>
```

Comment:
Where is `connectionString` from?

Reply:
Connection string is coming from json config. if you look into `CollectStatisticsDiagnostics` it takes an argument of action type named sqlOperation which goes to the server (TDSServer) and gets connection string from there.

Reply:
After looking deeper into this, server defines a localhost on the machine. it does not matter where we are. the TDSServer is defined with localhost in general with port number.  the connection string of config.json is used in some cases such as `ExecuteXmlReaderErrorTest`.

Reply:
check for `s_tcpConnStr` variable in the DiagnosticTests to see where we have used connection string from the config.json